inference/core/interfaces/camera/entities.py

```diff
@@ -96,4 +99,48 @@ def initialize_source_properties(self, properties: Dict[str, float]):
         pass
 
 
+class WebRTCVideoFrameProducer(VideoFrameProducer):
+    def __init__(self, to_inference_queue: deque, to_inference_lock: LockType):
+        self.to_inference_queue: deque = to_inference_queue
+        self.to_inference_lock: LockType = to_inference_lock
+        self._w: Optional[int] = None
+        self._h: Optional[int] = None
+        self._fps_buff = []
+        self._is_opened = True
+
+    def grab(self) -> bool:
+        return True
+
+    def retrieve(self) -> Tuple[bool, np.ndarray]:
+        while not self.to_inference_queue:
+            time.sleep(0.1)
```

Comment:
shall this be configurable?

Reply:
just a consideration - maybe threading queue to be used, then we could wait in read

Reply:
ok, I get the above is not feasible due to connecting async-non-async world on both ends of "pipe"